UPGRADE-4.2.md

```diff
@@ -16,6 +16,7 @@ Config
 
  * Deprecated constructing a `TreeBuilder` without passing root node information.
  * Deprecated `FileLoaderLoadException`, use `LoaderLoadException` instead.
+ * Deprecated using placeholder values (e.g. environment variables) with `cannotBeEmpty()`
```

Comment:
I think we should remove the word placeholder and use env var in all messages in this PR. It's interal details.